frontend/packages/dev-console/src/components/edit-deployment/utils/edit-deployment-utils.ts

```diff
@@ -280,6 +280,7 @@ export const convertDeploymentToEditForm = (
 ): EditDeploymentFormData => {
   const resourceType = getResourcesType(deployment);
   return {
+    formType: deployment.metadata.name ? 'edit' : 'create',
```

Comment:
@Lucifergene why are you adding this here when you are adding it in EditDeploymentForm as well?

Reply:
Actually, there are 2 different edge-cases as pointed out by @hemantsaini-7 

1. On clicking Reload button, and then changing the container image, results in an automatic change of Deployment Name
2. On switching to YAML view and coming back to the Form view, also results in the automatic change of Deployment Name

To fix this, I have added the `formType: deployment.metadata.name ? 'edit' : 'create'` in the `convertDeploymentToEditForm` function.

Reply:
I have removed the code changes in `EditDeploymentForm`